File: app/src/main/java/com/brooke/michael/filmbuff/ui/fragment/ThisMonthFragment.java
package com.brooke.michael.filmbuff.ui.fragment;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.brooke.michael.filmbuff.R;
import com.brooke.michael.filmbuff.adapter.MovieListRVAdapter;
import com.brooke.michael.filmbuff.rest.model.MovieList;
import com.brooke.michael.filmbuff.rest.service.RestClient;
import com.brooke.michael.filmbuff.rest.service.TheMovieDBService;

import butterknife.Bind;
import butterknife.ButterKnife;
import retrofit.Callback;
import retrofit.RetrofitError;
import retrofit.client.Response;

/**
 * Created by Michael on 06-Mar-16.
 */
public class ThisMonthFragment extends Fragment {

    @Bind(R.id.recycler_view)
    RecyclerView mRV;

    private MovieListRVAdapter mAdapter;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_this_month, container, false);

        ButterKnife.bind(this, rootView);
        mRV.setLayoutManager(new LinearLayoutManager(getActivity()));
        mRV.setHasFixedSize(true);
        mAdapter = new MovieListRVAdapter(getActivity());

        TheMovieDBService service = RestClient.getMovieDBClient();
        service.getCurrentMovies(new Callback<MovieList>() {
            @Override
            public void success(MovieList movieList, Response response) {
                mAdapter.setMovieList(movieList);
                mRV.setAdapter(mAdapter);
                mAdapter.notifyDataSetChanged();
            }

            @Override
            public void failure(RetrofitError error) {
                Toast.makeText(getActivity(), "API Error.", Toast.LENGTH_SHORT).show();
            }
        });

        return rootView;
    }
}

File: app/src/main/java/com/brooke/michael/filmbuff/ui/fragment/ThisMonthFragment.kt
package com.brooke.michael.filmbuff.ui.fragment

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.LinearLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.brooke.michael.filmbuff.R
import com.brooke.michael.filmbuff.adapter.MovieListRVAdapter
import com.brooke.michael.filmbuff.extensions.toast
import com.brooke.michael.filmbuff.rest.model.MovieList
import com.brooke.michael.filmbuff.rest.service.RestClient

import kotlinx.android.synthetic.main.fragment_this_month.*

import retrofit.Callback
import retrofit.RetrofitError
import retrofit.client.Response

class ThisMonthFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater!!.inflate(R.layout.fragment_this_month, container, false)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        recycler_view.layoutManager = LinearLayoutManager(activity)
        recycler_view.setHasFixedSize(true)

        RestClient.instance.getCurrentMovies(object : Callback<MovieList> {
            override fun success(movieList: MovieList, response: Response) {
                val mAdapter = MovieListRVAdapter(movieList)
                recycler_view.adapter = mAdapter
            }

            override fun failure(error: RetrofitError) {
                toast("API Error")
            }
        })
    }
}
